src/v2/models/AppState.js

```diff
@@ -35,21 +35,44 @@ export default Model.extend({
     factorProfile: {
       deps: ['factor'],
       fn (factor = {}) {
-        return factor.profile || {};
+        return factor && factor.profile || {};
```

Comment:
`factor = {}` in the function parameter would set `factor` to `{}` is it's undefined. why still add extra check?

Reply:
This was actually failing in the HeaderBeacon where we check for factor. I can move this check in there `factor && factor.factorType`

Reply:
do you mean this piece?
```
const factor = this.options.appState.get('factor');
    let className = 'undefined-user';
    if (factor) {
      className = getFactorData(factor.factorType).iconClassName;
    }
```
we actually already have `factorType` as derived prop, could you use that instead?
probably need to update `getFactorData` a bit by return `{}` if found nothing from `factorData`

Reply:
yeah, thats the place. I can use `factorType ` from appState